crates/swc_ecma_transforms_typescript/src/inline_enum.rs

```diff
@@ -0,0 +1,135 @@
+use std::{cell::RefCell, rc::Rc};
+
+use serde::{Deserialize, Serialize};
+use swc_atoms::JsWord;
+use swc_common::collections::AHashMap;
+use swc_ecma_ast::{
+    AssignExpr, ComputedPropName, Expr, Id, Lit, MemberExpr, MemberProp, Module, Script, Str, Tpl,
+};
+use swc_ecma_utils::ident::IdentLike;
+use swc_ecma_visit::{as_folder, noop_visit_mut_type, Fold, VisitMut, VisitMutWith};
+
+pub type TSEnumLit = Rc<RefCell<AHashMap<Id, AHashMap<JsWord, Lit>>>>;
+
+pub fn inline_enum(ts_enum_lit: TSEnumLit, inline_enum: InlineEnumConfig) -> impl Fold + VisitMut {
+    as_folder(InlineEnum {
+        ts_enum_lit,
+        inline_enum,
+    })
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum EnumConst {
+    Const,
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
+pub enum InlineEnumConfig {
+    Const(EnumConst),
+    All(bool),
+}
+
+impl Default for InlineEnumConfig {
+    fn default() -> Self {
+        Self::All(true)
```

Comment:
@kdy1 
I set "inline all as possible" as default option.
What do you think about it?

Reply:
I think it's good enough to be default

Reply:
There is a risk to make it default.
See https://www.typescriptlang.org/play?#code/KYOwrgtgBAglDeAoKsA0UBCiC+jEAo4BDAZyiJAE8BKAbQHIZ6BdKAXigEYBuA4sijVoAGVh06IgA